src/critnib/critnib.c

```diff
@@ -176,37 +182,68 @@ static inline unsigned slice_index(word key, sh_t shift) {
  * critnib_new -- allocates a new critnib structure
  */
 struct critnib *critnib_new(void) {
-    struct critnib *c = Zalloc(sizeof(struct critnib));
-    if (!c) {
+    umf_ba_linear_pool_t *pool_linear =
+        umf_ba_linear_create(0 /* minimal pool size */);
+    if (!pool_linear) {
         return NULL;
     }
 
-    c->mutex = util_mutex_create();
+    struct critnib *c =
+        umf_ba_linear_alloc(pool_linear, sizeof(struct critnib));
+    if (!c) {
+        goto err_destroy_pool_linear;
+    }
+
+    c->pool_linear = pool_linear;
+
+    void *mutex_ptr = umf_ba_linear_alloc(pool_linear, util_mutex_get_size());
+    if (!mutex_ptr) {
+        goto err_destroy_pool_linear;
+    }
+
+    c->mutex = util_mutex_init(mutex_ptr);
     if (!c->mutex) {
-        free(c);
-        return NULL;
+        goto err_destroy_pool_linear;
+    }
+
+    c->pool_nodes = umf_ba_create(sizeof(struct critnib_node));
```

Comment:
What are the sizes of `critnib_node` and `critnib_leaf`? I'm wondering if we should align them up to a cacheline to avoid false sharing.

Reply:
sizeof(struct critnib_node) = 144,
sizeof(struct critnib_leaf) = 16

Reply:
Ok, then we can probably leave it as is for now. We can consider this if we ever find that we need to optimize critnib.